roxie/ccd/ccdserver.cpp

```diff
@@ -7388,23 +7448,23 @@ class CRoxieServerDedupAllActivity : public CRoxieServerActivity
 
 class CRoxieServerDedupActivityFactory : public CRoxieServerActivityFactory
 {
-    bool compareAll;
-    bool keepLeft;
+    unsigned flags;
 
 public:
     CRoxieServerDedupActivityFactory(unsigned _id, unsigned _subgraphId, IQueryFactory &_queryFactory, HelperFactory *_helperFactory, ThorActivityKind _kind, IPropertyTree &_graphNode)
         : CRoxieServerActivityFactory(_id, _subgraphId, _queryFactory, _helperFactory, _kind, _graphNode)
     {
         Owned<IHThorDedupArg> helper = (IHThorDedupArg *) helperFactory();
-        compareAll = helper->compareAll();
-        keepLeft = helper->keepLeft();
+        flags = helper->getFlags();
     }
 
     virtual IRoxieServerActivity *createActivity(IRoxieSlaveContext *_ctx, IProbeManager *_probeManager) const
     {
-        if (compareAll)
-            return new CRoxieServerDedupAllActivity(_ctx, this, _probeManager, keepLeft);
-        else if (keepLeft)
+        if (flags & DFcompareall)
+            return new CRoxieServerDedupAllActivity(_ctx, this, _probeManager, flags & DFkeepleft);
+        else if (flags & DFkeepbest)
+            return new CRoxieServerDedupKeepBestActivity(_ctx, this, _probeManager, flags & DFkeepleft);
```

Comment:
Similar to hthor comment, since there's a lot in common, I wonder if not best to impl. best functionality in existing activity, rather than introduce new one.